feat: add error assertion options (#20247)
eslint/eslint: docs/src/integrate/nodejs-api.md

If a valid test case only uses the `code` property, it can optionally be provided as a string containing the code, rather than an object with a `code` key.

> You can optionally configure the following `assertionOptions`, these apply to all error assertions in that call:

Reviewer: ```suggestion
You can optionally configure the following `assertionOptions` that apply to all error assertions in the method call:
```

- `requireMessage` (boolean/"message"/"messageId", optional): If true, each `errors` block must check the expected error message, either via a string in the `errors` array, or via `message`/`messageId` in an errors object. `"message"`/`"messageId"` can be used to further limit the message assertions to the respective versions.

> - `requireLocation` (boolean, optional): If true, each `errors` block must be an array of objects, that each check all location properties `line`, `column`, `endLine`, `endColumn`, the later may be omitted, if the error does not contain them.

Reviewer: ```suggestion
- `requireLocation` (boolean, optional): If `true`, each `errors` block must be an array of objects, and each object must contain location properties `line`, `column`, `endLine`, and `endColumn`. Properties `endLine`, and `endColumn` may be omitted if the actual error does not contain them.
```
